trappy/ftrace.py

```diff
@@ -168,28 +168,21 @@ class definitions list.  Otherwise, register a class to parse
     def __populate_data(self, fin, cls_for_unique_word, window, abs_window):
         """Append to trace data from a txt trace"""
 
-        def contains_unique_word(line, unique_words=cls_for_unique_word.keys()):
-            for unique_word in unique_words:
-                if unique_word in line:
-                    return True
-            return False
-
         actual_trace = itertools.dropwhile(self.trace_hasnt_started(), fin)
         actual_trace = itertools.takewhile(self.trace_hasnt_finished(),
                                            actual_trace)
 
         for line in actual_trace:
-            if not contains_unique_word(line):
-                self.lines += 1
-                continue
+            trace_class = None
             for unique_word, cls in cls_for_unique_word.iteritems():
                 if unique_word in line:
                     trace_class = cls
                     if not cls.fallback:
                         break
             else:
                 if not trace_class:
-                    raise FTraceParseError("No unique word in '{}'".format(line))
+                    self.lines += 1
+                    continue
```

Comment:
The assumption here is that we loop on all the events we know about and we are able to parse, if none matches their unique word that means we can skip this line while still counting for the trace line we parsed.

That's why we replaced the exception with a continue.

Moreover, I think the previous `raise` was already dead-code. Indeed, we was checking for a `unique_word` match using `contains_unique_word` right before the loop, thus if we enter the loop we will certainly set `trace_class` to some (hopefully not null) `cls`.